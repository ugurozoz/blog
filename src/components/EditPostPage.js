import React from "react";
import { Link } from "react-router-dom";

const EditPostPage = () => <div>Edit Page</div>;

export default EditPostPage;
