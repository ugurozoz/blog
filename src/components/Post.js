import React from "react";
import moment from "moment";

export default class Post extends React.Component {   
  constructor(props) {
    super(props);

    this.state = {
      title: props.post ? props.post.title : "",
      body: props.post ? props.post.body : "",
      createdAt: props.post ? moment(props.post.createdAt) : moment(),
      error: ""
    };    
  }
  render() {
    
    return (
      <div>
        <div className="page-header">{this.state.title}</div>
        <div className="page-body">{this.state.body}</div>
      </div>
    )
  }
}