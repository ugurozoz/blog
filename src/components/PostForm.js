import React from "react";
import moment from "moment";

export default class PostForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      title: props.post ? props.post.title : "",
      body: props.post ? props.post.body : "",
      createdAt: props.post ? moment(props.post.createdAt) : moment(),
      error: ""
    };
  }

  onBodyChange = e => {
    const body = e.target.value;
    this.setState(() => ({ body }));
  };

  onTitleChange = e => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };

  onSubmit = e => {
    e.preventDefault();
    if (!this.state.title || !this.state.body) {
      this.setState(() => ({
        error: "Please provide title and body"
      }));
    } else {
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        title: this.state.title,
        body: this.state.body,
        createdAt: this.state.createdAt.valueOf()
      });
    }
  };

  render() {
    return (
      <div>
        <form className="form" onSubmit={this.onSubmit}>
          {this.state.error && <p>{this.state.error}</p>}
          <input
            type="text"
            placeholder="title"
            autoFocus
            value={this.state.title}
            onChange={this.onTitleChange}
            className="text-input"
          />
          <textarea
            placeholder="body"
            value={this.state.body}
            onChange={this.onBodyChange}
            className="textarea"
          ></textarea>
          <button className="button button--add">Add Blog</button>
        </form>
      </div>
    );
  }
}
