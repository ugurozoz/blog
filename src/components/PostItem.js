import React from "react";
import { Link } from "react-router-dom";

const PostItem = ({ id, title, body, createdAt }) => (
  <Link className="post-item" to={`edit/${id}`}>
    <div className="post">
      <strong>{title}</strong>
    </div>
  </Link>
);

export default PostItem;
