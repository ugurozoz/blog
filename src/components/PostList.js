import React from 'react';

export const PostList = () => (  
    <div>Post List</div>  
)

export default PostList