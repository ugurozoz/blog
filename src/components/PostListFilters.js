import React from "react";
import { connect } from "react-redux";
import DateRangePicker from "react-dates";
import "react-dates/lib/css/_datepicker.css";
import {
  setTextFilter,sortByName,sortByDate
} from "../actions/filters";

export class PostListFilters extends React.Component {
  
  onTextChange = e => {
    this.props.setTextFilter(e.target.value);
  };


  onSortChange = e => {
    if (e.target.value === "date") {
      this.props.sortByDate();
    } else if (e.target.value === "name") {
      this.props.sortByName();
    }

  }



  render() {
    return (
      <div className="post-filters">
    <input
              placeholder="Search Posts"
              className="text-input"
              type="text"
              value={this.props.filters.text}
              onChange={this.onTextChange}
        />
      <select
              className="select"
              value={this.props.filters.sortBy}
              onChange={this.onSortChange}
            >
              <option value="date">Date</option>
              <option value="name">Name</option>
            </select>
</div>

    )
  }
  
  
}

const mapStateToProps = state => ({
  filters: state.filters
});

const mapDispatchToProps = dispatch => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByName: () => dispatch(sortByName())

  
});


export default connect(mapStateToProps,mapDispatchToProps)(PostListFilters);
