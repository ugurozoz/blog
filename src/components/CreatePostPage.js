import React from "react";
import { connect } from "react-redux";
import { startCreatePost } from "../actions/posts";
import PostForm from "./PostForm";

export class CreatePostPage extends React.Component {
  onSubmit = post => {
    this.props.startCreatePost(post);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <div>Create Post</div>
        <PostForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startCreatePost: post => dispatch(startCreatePost(post))
});

export default connect(
  undefined,
  mapDispatchToProps
)(CreatePostPage);
