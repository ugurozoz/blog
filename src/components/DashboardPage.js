import React from "react";
import PostList from ".//PostList";
import PostListFilters from "./PostListFilters";
import { Link } from "react-router-dom";

const DashboardPage = () => {
  return (
    <div>
      <div>Dashboard Page</div>
      <div>
        <Link to="/create">Create Post</Link>
      </div>
      <PostListFilters />
      <PostList />
    </div>
  );
};

export default DashboardPage;
