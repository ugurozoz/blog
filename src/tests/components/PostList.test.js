import React from 'react';
import { shallow } from 'enzyme';
import { PostList } from "../../components/PostList";


test("should render Post List page", () => {
  const wrapper = shallow(<PostList />);
  expect(wrapper).toMatchSnapshot();
});
