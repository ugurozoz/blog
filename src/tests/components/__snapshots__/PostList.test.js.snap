// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Post List page 1`] = `
<div>
  Post List
</div>
`;
